src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1499,6 +1510,37 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 		}
 	}
 
+	private async explodeConflictingFolderEdit(edit: ResourceFileEdit): Promise<ResourceFileEdit[]> {
+		if (!edit.oldResource || !edit.newResource) {
+			return [edit];
+		}
+
+		if (!await this.fileService.exists(edit.newResource)) {
+			return [edit];
+		}
+
+		const resolvedOldResource = await this.fileService.resolve(edit.oldResource);
+		const resolvedNewResource = await this.fileService.resolve(edit.newResource);
+
+		//TODO: what to do here!?
+		if (!resolvedOldResource.isDirectory || !resolvedNewResource.isDirectory) {
+			return [edit];
+		}
+
+		const edits: ResourceFileEdit[] = [];
+
+		const newEdits = resolvedOldResource.children?.map(child => new ResourceFileEdit(child.resource, URI.joinPath(edit.newResource!, child.name)));
+
+		for (const newEdit of newEdits!) {
+			edits.push(...await this.explodeConflictingFolderEdit(newEdit));
+		}
+
+		// Since all children of the folder are now moved individually, we have to delete the folder
+		edits.push(new ResourceFileEdit(edit.oldResource, undefined));
```

Comment:
In here, the deletion will be recursively added.
But as I said in the text, this causes issues.